azure-pipelines.yml

```diff
@@ -113,18 +113,57 @@ stages:
         env:
           TeamName: $(_TeamName)
         condition: and(succeeded(), in(variables['_SignType'], 'real', 'test'), eq(variables['Agent.Os'], 'Windows_NT'))
-    
+
       - task: PowerShell@2
-        displayName: Sign artifacts and Package Microsoft.Spark.Worker
+        displayName: Sign artifacts
         inputs:
           filePath: eng\common\build.ps1
-          arguments: -restore -sign -publish
+          arguments: -restore -sign
                      -c $(buildConfiguration)
                      -ci
                      $(_OfficialBuildIdArgs)
                      /p:DotNetSignType=$(_SignType)
                      /p:SparkPackagesDir=$(ArtifactPath)\BuildArtifacts\artifacts\packages
                      /p:SparkWorkerPublishDir=$(ArtifactPath)\Microsoft.Spark.Worker
+
+      - task: PublishBuildArtifacts@1
+        inputs:
+          pathtoPublish: '$(ArtifactPath)'
+          artifactName:  Microsoft.Spark.Binaries
+
+  # The "Publish" stage is separated out from the "Sign" stage because we need to install powershell module
+  # to zip files correctly for macOS; installing the module is not allowed in NetCoreInternal-Pool.
+  - ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+    - job: Publish
+      dependsOn:
+        - Sign
+      displayName: Publish Artifacts
+      pool: Hosted VS2017
+
+      variables:
+        ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'PullRequest')) }}:
+          _OfficialBuildIdArgs: /p:OfficialBuildId=$(BUILD.BUILDNUMBER)
+
+      steps:
+      # The following module needs to be installed to zip files correctly for macOS.
+      - powershell: Install-Module -Name Microsoft.PowerShell.Archive -Scope CurrentUser -Force -AllowClobber -Verbose -MinimumVersion 1.2.5
+        displayName: Install Microsoft.PowerShell.Archive
```

Comment:
I think it's more obvious to put this here instead inside the `package-worker.ps1`.